packages/google_sign_in/google_sign_in/lib/google_sign_in.dart

```diff
@@ -211,12 +206,12 @@ class GoogleSignIn {
   // Future that completes when we've finished calling `init` on the native side
   Future<void> _initialization;
 
-  Future<GoogleSignInAccount> _callMethod(String method) async {
+  Future<GoogleSignInAccount> _callMethod(Function method) async {
     await _ensureInitialized();
 
-    final Map<String, dynamic> response =
-        await channel.invokeMapMethod<String, dynamic>(method);
-    return _setCurrentUser(response != null && response.isNotEmpty
+    final dynamic response = await method();
```

Comment:
what would happen if the `method` is not an async function.

Reply:
Good call! It seems to work, though, check this [DartPad](https://dartpad.dartlang.org/f062d40311b09b39c6bdef925137653e)!